### PR TITLE
KAFKA-4115: Increasing the heap settings for connect-distributed script

### PR DESCRIPTION
**Pull request by wicknicks**
Signed-off-by: Arjun Satish <arjun@confluent.io>

### CODE REVIEW

**Comment by wicknicks** [comment]
@rhauch and @kkonstantine could you please help review these new settings. Thanks.

File: bin/connect-distributed.sh [hunk 1]
```diff
@@ -26,6 +26,10 @@ if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
     export KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/connect-log4j.properties"
 fi
 
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+  export KAFKA_HEAP_OPTS="-Xms256M -Xmx2G"
+fi
+
```
Comment (rhauch) [inline bin/connect-distributed.sh:32]: Should we also do this in `bin/connect-standalone.sh`?

Comment (rhauch) [reply bin/connect-distributed.sh:32]: But, yes, IMO these are the right settings to make and the correct way to do it.

**Comment by asfgit** [comment]
FAILURE
 7959 tests run, 5 skipped, 2 failed.
--none--

File: bin/connect-distributed.sh (hunk 1 above)
Comment (wicknicks) [reply bin/connect-distributed.sh:32]: Yeah, it should be better to keep both these settings consistent. Let me update this PR. Thanks, @rhauch.

**Comment by asfgit** [comment]
SUCCESS 
 8067 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Review by rhauch — APPROVED** [review]
LGTM

**Comment by asfgit** [comment]
FAILURE
 7959 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7959 tests run, 5 skipped, 1 failed.
--none--

**Comment by wicknicks** [comment]
retest this please

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8067 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by wicknicks** [comment]
retest this please

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8067 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8067 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7959 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7959 tests run, 5 skipped, 1 failed.
--none--

**Comment by wicknicks** [comment]
retest this please

**Comment by asfgit** [comment]
SUCCESS 
 8071 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7963 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8071 tests run, 5 skipped, 0 failed.
--none--

**Comment by wicknicks** [comment]
retest this please

**Comment by asfgit** [comment]
SUCCESS 
 8071 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7963 tests run, 5 skipped, 1 failed.
--none--

File: bin/connect-distributed.sh (hunk 1 above)
Comment (bencoughlan) [reply bin/connect-distributed.sh:32]: Should this be accompanied with a log message on output. 

i.e. 

"No JVM memory settings set: Sticking to the Default of Min : 256 mb Max: 2048mb."

The JVM memory limitations/settings weren't super clear to me when I started provisioning clusters.

**Comment by asfgit** [comment]
SUCCESS 
 8071 tests run, 5 skipped, 0 failed.
--none--

File: bin/connect-distributed.sh (hunk 1 above)
Comment (wicknicks) [reply bin/connect-distributed.sh:32]: @bencoughlan: we don't want to output anything in the shell scripts (as free text in shell scripts might corrupt the application log stream). An alternative would be to add some documentation on what the default heap settings are, and how we can change them while running the connect scripts. Thoughts?

Comment (kkonstantine) [reply bin/connect-distributed.sh:32]: Adding here that the jvm options are now printed by the Worker's logger right at the beginning of the execution, along with the rest of the options passed to the jvm. 

Not exactly the notification you describe below @bencoughlan, but the information is there.

Comment (wicknicks) [reply bin/connect-distributed.sh]: A quick addition to my comment: All the jvm args are already being printed when the connect workers start. It should be the first or second log message in the worker. Hope this helps.

**Comment by wicknicks** [comment]
@ewencp: could you please merge this PR. The failing test has nothing to do with the changes in this PR (as confirmed by @rajinisivaram). Thanks!

File: bin/connect-distributed.sh (hunk 1 above)
Comment (bencoughlan) [reply bin/connect-distributed.sh:32]: I mean, yeah, that sounds like a good idea. 

even if it was in the properties file be default over each value 

#Default = X if not set. 

etc. 

Does that make sense? 

This JVM setting, in particular, wasn't overly obvious.

**Review by ewencp — APPROVED** [review]
LGTM

**Pull request closed without merging**